SCons/Scanner/LaTeX.py

```diff
@@ -432,7 +433,9 @@ def scan_recurse(self, node, path=()):
                 # recurse down
                 queue.extend(self.scan(n, inc_subdir))
 
-        return [pair[1] for pair in sorted(nodes)]
+        # Don't sort on a tuple where the second element is an object, just
+        # use the first element of the tuple which is the "sort_key" value
+        return [pair[1] for pair in sorted(nodes, key=lambda n: n[0])]
```

Comment:
Frankly, if `nodes` what just the File nodes themselves, this sort could be `sorted(nodes, key=self.sort_key)`. I was trying to avoid too many changes in one PR that did not address the specific issue.

Reply:
Ahh that's a good point, though for this issue it wouldn't have affected the win32 vs posix sort order.

I'm not sure more changes are warranted at this point, as you said they're not really related to the fix.
Are you ok with the changes I've made? 
If so, I'll go ahead and merge.